sql-parser/src/main/antlr/SqlBase.g4

```diff
@@ -215,6 +215,7 @@ valueExpression
     | left=valueExpression operator=(PLUS | MINUS) right=valueExpression             #arithmeticBinary
     | left=valueExpression CONCAT right=valueExpression                              #concatenation
     | valueExpression CAST_OPERATOR dataType                                         #doubleColonCast
+    | dataType stringLiteral                                                         #fromStringLiteralCast
```

Comment:
Should we restrict this to simple types? (exclude arrays, object definition).

E.g. from the postgres docs:

> To avoid syntactic ambiguity, the type 'string' syntax can only be used to specify the type of a simple literal constant. Another restriction on the type 'string' syntax is that it does not work for array types; use :: or CAST() to specify the type of an array constant.

SQL Standard also doesn't require us to allow it for all types.